src/ngraph/op/reshape.cpp

```diff
@@ -500,6 +500,6 @@ bool op::v1::Reshape::evaluate(const HostTensorVector& outputs, const HostTensor
         }
         outputs[0]->set_shape(output_shape);
     }
-    const AxisVector order = get_default_order(outputs[0]->get_shape());
+    const AxisVector order = get_default_order(inputs[0]->get_shape());
```

Comment:
@rkimball Is this correct? Then it means there was a long standing bug?

Reply:
Yes, the input for v0 Reshape is the input axis order, not the output order. When I enabled the EVAL backend it was segfaulting because the output shape was smaller than the input shape so the axis order vector was too small and we read past the end. With this change unit tests pass and 100% less segfaults.